src/view/components/EditEntry/index.tsx

```diff
@@ -0,0 +1,218 @@
+    const isInfo = color === 'INFO';
+    const isPrimary = color === 'PRIMARY';
+
+    return (
+      <Col style={Styles.button}>
+        <Button block info={isInfo} primary={isPrimary} onPress={onPress}>
+          <Text>{text}</Text>
+        </Button>
+      </Col>
+    );
+  }
+
+  private renderHeader() {
+    return (
+      <Grid>
+        <Row style={Styles.headerRow}>
+          <H1>
+            {this.props.farmer.firstName} {this.props.farmer.lastName}
+          </H1>
+        </Row>
+        <Row style={Styles.headerRow}>
+          <Text style={Styles.header}>
+            {/* @TODO Change this to take date only */}
+            {this.props.milkEntry.datetime}
+          </Text>
+        </Row>
+        <Row style={Styles.headerRow}>
+          <Text style={Styles.header}>
+            {/* @TODO Change this to take time only */}
+            {this.props.milkEntry.datetime} 
+          </Text>
+        </Row>
+      </Grid>
+    );
+  }
+
+  private formatEditRow(label: string, value: number | string, onChangeText: any) {
+    return (
+      <Grid>
+        <Row>
+          <Col>
+            <Text>{label}</Text>
+          </Col>
+          <Col>
+          <Item>
+            <Input onChangeText={onChangeText}>
+              <Text>{value}</Text>
+            </Input>
+          </Item>
+          </Col>
+        </Row>
+      </Grid>
+    );
+  }
+
+  private renderEditFields() {
+    return (
+      <View style={Styles.editView}>
+        {this.formatEditRow('Amount (L)', this.props.milkEntry.amountOfProduct, this.onAmountChange)}
+        {this.formatEditRow('Quality', this.props.milkEntry.quality, this.onQualityChange)}
+        {this.formatEditRow('Rate (UGX)', this.props.milkEntry.costPerUnit, this.onRateChange)}
+      </View>
+    );
+  }
+
+  /**
+   * Render method for EditEntry
+   */
+  public render() {
+    return(
+      <Content padder style={{ backgroundColor: 'white' }}>
+      <List>
+        <ListItem>
+          {this.renderHeader()}
+        </ListItem>
+      </List>
+      {this.renderEditFields()}
+      <Grid>
+        <Row style={Styles.buttonRow}>
+          {this.renderCancelButton()}
+          {this.renderSaveButton()}
+        </Row>
+      </Grid>
+    </Content>
+    );
+  }
+}
+
+const EditEntryPage = new Composer<NestedPropsType>(EditEntry).page;
+
+const mapStateToProps: MapStateToProps<StorePropsType, OwnPropsType, State> = (state) => {
+  // @TODO replace 'fakeFarmerUUID' with the active farmer uuid
+  const farmerRow = state.farmer.rows.find(r => r.uuid === 'fakeFarmerUUID');
+  if (farmerRow === undefined) {
+    throw new Error('Error: could not locate farmer: ' + 'fakeFarmerUUID');
+  }
+  // @TODO replace 'fakeMilkEntryUUID' with the active MilkEntry uuid    
+  const milkEntryRow = state.milk.rows.find(r => r.uuid === 'fakeMilkEntryUUID');
+  if (milkEntryRow === undefined) {
+    throw new Error('Error: could not locate MilkEntry: ' + 'fakeMilkEntryUUID');
+  }
```

Comment:
do we want all these references to fake data in our master. Are we keeping it?

Reply:
We don't, you're right.